tests/common/test_logger.rs

```diff
@@ -0,0 +1,66 @@
+use std::ops::Deref;
```

Comment:
Why did we need to bring in this file rather than use the crate like before?

Reply:
This one is interesting and I may not fully understand it; so if I'm getting something wrong it'd be great if you could explain it to me. It has to do with thread-local variables, which I don't fully understand.

So the external crate was capturing log messages in a `Vec` that was a thread-local variable. When our code went multi-threaded, my understanding is that this meant that the master thread running the tests was no longer able to access the `Vec`s that were being built up on the child threads it spawned. So I basically took the external crate and modified it so that the logger is appending to a global `Arc<Mutex<Vec<CapturedLog>>>` so we can see all the log messages from every thread on any thread.

But again, I may be misunderstanding how thread-local variables work.

Reply:
See [the external crate's code](https://github.com/psFried/test-logger/blob/master/src/lib.rs)

Reply:
Interesting.. It seems to me like that crate will only ever work if all the logging happens in one thread. Doing it this way seems fine, though I think then a comment at the top of the file would be useful :)

Reply:
I'll continue to think about a better way to do this too, though. And if you have any ideas let me know. As the current solution is what requires us to run the tests with `--test-threads=1`, which would be nice to get rid of if we can.

Reply:
I think sadly the solution here is to explicitly pass around a logger using something like `slog`, which is unclear is worth the change. I think it's possibly worth doing that in a separate PR.

Reply:
Another alternative here is to switch to the new `tokio-trace`, which has thread-local logging. You would basically set the logger ("subscriber" in the `tokio-trace` world) in each thread you spawn to one you create for that test.